src/Riok.Mapperly/Descriptors/MappingBuilders/EnumToStringMappingBuilder.cs

```diff
@@ -18,6 +22,43 @@ public static class EnumToStringMappingBuilder
 
         // to string => use an optimized method of Enum.ToString which would use slow reflection
         // use Enum.ToString as fallback (for ex. for flags)
-        return new EnumToStringMapping(ctx.Source, ctx.Target, ctx.SymbolAccessor.GetAllFields(ctx.Source));
+        return BuildEnumToStringMapping(ctx);
+    }
+
+    private static EnumToStringMapping BuildEnumToStringMapping(MappingBuilderContext ctx)
+    {
+        var ignoredSourceMembers = ctx.Configuration.Enum.IgnoredSourceMembers.ToHashSet(SymbolTypeEqualityComparer.FieldDefault);
+        var explicitMappings = BuildExplicitValueMappings(ctx);
+
+        EnumMappingDiagnosticReporter.AddUnmatchedSourceIgnoredMembers(ctx, ignoredSourceMembers);
+        return new EnumToStringMapping(ctx.Source, ctx.Target, ctx.SymbolAccessor.GetAllFields(ctx.Source), explicitMappings);
+    }
+
+    private static IReadOnlyDictionary<IFieldSymbol, ExpressionSyntax> BuildExplicitValueMappings(MappingBuilderContext ctx)
+    {
+        var explicitMappings = new Dictionary<IFieldSymbol, ExpressionSyntax>(SymbolEqualityComparer.Default);
+        foreach (var (source, target) in ctx.Configuration.Enum.ExplicitMappings)
+        {
+            var sourceField = ctx.SymbolAccessor.GetEnumField(source.ConstantValue)!;
```

Comment:
I'm not sure if instead of `GetEnumField` it would make sense to collect the source fields into a `Dictionary<object, IFIeldSymbol>` to access each field by its constant value (O(n)) in constant time instead of looping each time with `FirstOrDefault` (O(n^2)). The same applies to other callers of `GetEnumField`.

Reply:
Yes, I agree. I have updated the approach to collect the source fields into a Dictionary<object, IFieldSymbol>